assets/src/js/media_archive_editor_view.js

```diff
@@ -0,0 +1,63 @@
+import {createRoot} from '@wordpress/element';
+import ArchivePicker from '../js/Components/ArchivePicker/ArchivePicker';
+
+const frame = wp.media.view.MediaFrame.Select;
+
+(function() {
+  wp.media.view.MediaFrame.Select = frame.extend({
+    initialize() {
+      frame.prototype.initialize.apply(this, arguments);
+
+      const State = wp.media.controller.State.extend({
+        insert() {
+          this.frame.close();
+        },
+      });
+
+      this.states.add([
+        new State({
+          id: 'mediaArchive',
+          search: false,
+          title: 'Media Archive',
+        }),
+      ]);
+
+      this.on('content:render:mediaArchive', this.renderMediaArchive, this);
+    },
+
+    browseRouter(routerView) {
+      routerView.set({
+        upload: {
+          text: wp.media.view.l10n.uploadFilesTitle,
+          priority: 20,
```

Comment:
I think just defining 1,2 or 3 it's fine

Reply:
Yes it would be if i was sure of the behaviour when the view is rendered. I couldn't find anything specific but most of the resources and code snippets suggested this, stating possible overriding default behaviour issues or rendering and re-rendering issues and this being like a safe value.